pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -107,7 +108,8 @@ func (p *ConfigurationEditFlags) addSharedFlags(command *cobra.Command) {
 		"keep the running image for the service constant when not explicitly specifying "+
 			"the image. (--no-lock-to-digest pulls the image tag afresh with each new revision)")
 	// Don't mark as changing the revision.
-
+	command.Flags().StringVar(&p.ServiceAccountName, "service-account-name", "-", "Service account name. To unset, specify \"-\".")
```

Comment:
any particular reason we are not using default as `-` instead of `nil`?

Reply:
I've changed the default value to an empty string.

Reply:
thanks!